### ISSUE TITLE
L1 memory error - deepseek OCR

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-xla#2768]
### Describe the bug 

- Deepseek OCR failing due to `Statically allocated circular buffers on core range [(x=0,y=0) - (x=7,y=7)] grow to 1732864 B which is beyond max L1 size of 1499136 B` - [jan9_z_deepseek_ocr.log](https://github.com/user-attachments/files/24543550/jan9_z_deepseek_ocr.log)
- Model size(approx) -  3.336B - [jan9_deepseek_ocr_size.log](https://github.com/user-attachments/files/24522613/jan9_deepseek_ocr_size.log)

### Root cause analysis

- From the error trace, it's clear that ttnn.softmax hits L1 error.
- Further slicing originates from [scaled_dot_product_attention](https://github.com/tenstorrent/tt-forge-models/blob/1595465207d288d48aacb3f0f2b81eb28363d26d/deepseek/deepseek_ocr/pytorch/src/deepencoder.py#L743C37-L743C65) present in attn of 3rd block of sam model confirmed by below experiments. 
  - Case 1: Return tensor before SDPA - PASSED - [deepencoder.py](https://github.com/user-attachments/files/24562623/deepencoder.py), [jan12_deepseek_ocr_case1.log](https://github.com/user-attachments/files/24562622/jan12_deepseek_ocr_case1.log), [loader.py](https://github.com/user-attachments/files/24562621/loader.py)
  - Case 2: Return tensor after SDPA - L1 error - [deepencoder.py](https://github.com/user-attachments/files/24562627/deepencoder.py), [jan12_deepseek_ocr_case2.log](https://github.com/user-attachments/files/24562626/jan12_deepseek_ocr_case2.log), [loader.py](https://github.com/user-attachments/files/24562628/loader.py)
 
- But unable to replicate error with sanity. SDPA sanity hangs - [jan12_sdpa_exp1.log](https://github.com/user-attachments/files/24562644/jan12_sdpa_exp1.log)
- Expanded the sanity but adding some more ops , but issue persists - [jan12_sdpa_exp2.log](https://github.com/user-attachments/files/24562656/jan12_sdpa_exp2.log)
- From the TTNN graph of exp1 & TTNN repro of sam model, it's clear that input shape is (1, 12, 4096, 4096) - [jan9_z_deepseek_ocr_sam_emitpy_run_details.log](https://github.com/user-attachments/files/24562789/jan9_z_deepseek_ocr_sam_emitpy_run_details.log)

- But softmax sanity for tt-metal passed for this shape. [jan9_softmax.log](https://github.com/user-attachments/files/24562787/jan9_softmax.log)
- So unable to replicate  error with either softmax/SDPA/SDPA + few preceding ops.

### TT-XLA Repro

```
git checkout kkannan/jan12_deepseek_ocr_L1
git submodule update --init —recursive

# model run
pytest tests/runner/test_models.py::test_all_models_torch[deepseek/deepseek_ocr/pytorch-deepseek_ocr-single_device-inference] -svv --runxfail

# cases run 
# replace the attached script contents in corresponding files and rerun above command 

# exp run 
pytest tests/torch/ops/test_sdpa.py -k "exp1" -svv 
pytest tests/torch/ops/test_sdpa.py -k "exp2" -svv

```

### TT-METAL repro

```
git checkout kkannan/jan9_deepseek_ocr_softmax_l1_error
git submodule update --init —recursive

pytest tests/ttnn/unit_tests/operations/softmax/test_softmax.py -svv 
```
